content/themis/languages/nodejs/installation.md

```diff
@@ -89,3 +93,40 @@ you can manually build and install the latest version of Themis from source code
     ```bash
     npm install /path/to/jsthemis.tgz
     ```
+
+## Matching OpenSSL versions
+
+JsThemis is a simple addon to Nodejs that provides bindings to the Themis library, which is itself installed as a shared library in the system. In turn, Themis depends on OpenSSL installed in the system.
+
+However, [Nodejs often comes with its own OpenSSL version](https://github.com/nodejs/TSC/blob/main/OpenSSL-Strategy.md) included in the binary. This is a problem for JsThemis, because due to how linkage works, some OpenSSL functions will be linked directly from Nodejs and the others will come from the system's OpenSSL. If these OpenSSL versions are not the same, JsThemis will not work correctly or crash.
+
+```
+                          +-----> Node openssl
+                          |
+jsthemis ----> libthemis -+
+                          |
+                          +-----> System openssl
+```
+
+For these reasons, before installing JsThemis, it is critical to ensure that the Nodejs and system OpenSSL versions are the same. To do so, you can check the output of these commands:
+
+```bash
+$ openssl version
+OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)
+$ node -e "console.log(process.versions['openssl'])"
+3.0.2+quic
+```
+
+If the semantic versions are the same, you are good to go! If they differ, however, there are a few ways to resolve the situation, but there is no easy answer:
+
+1. Choose the Nodejs version with the OpenSSL that matches with your systems' one or that has no OpenSSL built in (there may be versions of Nodejs that use shared OpenSSL).
+
+   For example, Ubuntu Jammy Jellyfish (22.04) has OpenSSL 3.0.2, so the only Nodejs versions suitable for it are v18.0 and v18.1. On the other hand, Ubuntu Focal Fossa (20.04) has OpenSSL 1.1.1, which is compatible with Node LTS v14 and v12. However, be aware that OpenSSL 1.1.1 [will be unsupported after September 2023](https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/).
+
+2. Try another distribution with OpenSSL that matches Nodejs.
+
+3. Try to install OpenSSL version that matches your Nodejs. Probably, it will require building OpenSSL from the sources. Then, you will have to rebuild [Themis and JsThemis from sources](#building-latest-version-from-source). Don't forget to [specify the path](../../../installation/installation-from-sources/#cryptographic-backends) to the new OpenSSL.
```

Comment:
```suggestion
3. Try to install OpenSSL version that matches your Node.js. Probably, it will require building OpenSSL from the sources. Then, you will have to rebuild [Themis and JsThemis from sources](#building-latest-version-from-source). Don't forget to [specify the path](../../../installation/installation-from-sources/#cryptographic-backends) to the new OpenSSL.
```